replaceholder.js

```diff
@@ -1,23 +1,64 @@
-(function ($) {
-    $.fn.replaceholder = function () {
+(function (global, factory) {
+    'use strict';
+
+    if (typeof define === 'function' && define.amd) {
+        // AMD require.js
+        define('replaceholder', ['jquery'], function ($) {
+            $.fn.replaceholder = factory();
+        });
+    } else if (typeof exports !== 'undefined') {
+        // Node.js CommonJS
+        require('jquery').fn.replaceholder = factory();
+    } else if (typeof global.$ === 'function' && typeof global.$.fn === 'object') {
+        // Global scope
+        global.$.fn.replaceholder = factory();
+    }
+}(this, function () {
+    'use strict';
+
+    // options argument is an object that may contain:
+    // src: 'newImage.png' // required image url
+    // bg: false           // replaces src attribute (false) or background-image style (true) of the element
+    // doneClass: 'string' // class or classes that will be added to the element when image is loaded
+    // failClass: 'string' // class or classes that will be added to the element if image can't be loaded
+    // replaceClass: false // doneClass and failClass will rewrite (true) or be added (false) current value
+    return function (options) {
+        options = options || {};
         return this.each(function () {
-            var img = $(this);
-            var src = img.data('src');
-            var type = img.data('type');
+            var img = this;
+            var src = options.src || img.getAttribute('data-src');
```

Comment:
jQuery independent would be cool, we could make 2 versions.
Didn't know that .data was that slow(https://jsperf.com/getattribute-vs-data-3)